test/test_torchscript_consistency.py

```diff
@@ -304,85 +304,75 @@ def test_perf_biquad_filtering(self):
         _test_torchscript_functional(F.lfilter, waveform, a, b)
 
 
-RUN_CUDA = torch.cuda.is_available()
-print("Run test with cuda:", RUN_CUDA)
+def _assert_transforms_consistency(transform, tensor, device):
+    tensor = tensor.to(device)
+    transform = transform.to(device)
+    ts_transform = torch.jit.script(transform)
+    output = transform(tensor)
+    ts_output = ts_transform(tensor)
 
+    torch.testing.assert_allclose(ts_output, output)
 
-def _test_script_module(f, tensor, *args, **kwargs):
 
-    py_method = f(*args, **kwargs)
-    jit_method = torch.jit.script(py_method)
-
-    py_out = py_method(tensor)
-    jit_out = jit_method(tensor)
-
-    torch.testing.assert_allclose(jit_out, py_out)
-
-    if RUN_CUDA:
-
-        tensor = tensor.to("cuda")
-
-        py_method = py_method.cuda()
-        jit_method = torch.jit.script(py_method)
+class _TransformsTestMixin:
+    """Implements test for Transforms that are performed for different devices"""
+    device = None
 
-        py_out = py_method(tensor)
-        jit_out = jit_method(tensor)
+    def _assert_conssitency(self, transform, tensor):
+        _assert_transforms_consistency(transform, tensor, self.device)
 
-        torch.testing.assert_allclose(jit_out, py_out)
-
-
-class TestTransforms(unittest.TestCase):
     def test_Spectrogram(self):
         tensor = torch.rand((1, 1000))
-        _test_script_module(torchaudio.transforms.Spectrogram, tensor)
+        self._assert_conssitency(T.Spectrogram(), tensor)
```

Comment:
Were you trying to reduce the use of classes/methods in each test?

Reply:
Yes, that's why the actual implementation is placed outside of the class and this helper method is a syntax sugar to pass `self.device` automatically. This way, even if we want to get rid of class structure, it's easy.